common/persistence/dataInterfaces.go

```diff
@@ -768,6 +768,10 @@ type (
 		Condition int64
 		RangeID   int64
 
+		// for fork run (we need to make sure the forkRun hasn't been deleted)
+		ForkRunID          string
+		ForkRunNextEventID int64
```

Comment:
probably should also include ForkRunLastWriteVersion

Reply:
OK

Reply:
Confirmed offline that what we need is PrevRunVersion/PrevRunState for the current run to assert on current workflow record